sdk/storage/Azure.Storage.Queues/src/QueueClientSideEncryptionOptions.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Storage.Cryptography;
+
+#pragma warning disable SA1402  // File may only contain a single type
+
+namespace Azure.Storage.Queues.Specialized
+{
+    /// <summary>
+    /// Contains Queues-specific options for client-side encryption.
+    /// </summary>
+    public class QueueClientSideEncryptionOptions : ClientSideEncryptionOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QueueClientSideEncryptionOptions"/> class.
+        /// </summary>
+        /// <param name="version">The version of clientside encryption to use.</param>
+        public QueueClientSideEncryptionOptions(ClientSideEncryptionVersion version) : base(version)
+        {
+        }
+
+        /// <summary>
+        /// Event when failure to decrypt a message occurs.
+        /// </summary>
+        public event EventHandler<ClientSideDecryptionFailureEventArgs> DecryptionFailed;
+
+        internal bool UsingDecryptionFailureHandler => DecryptionFailed.GetInvocationList().Length > 0;
+
+        internal void OnDecryptionFailed(object message, Exception e)
+        {
+            DecryptionFailed?.Invoke(this, new ClientSideDecryptionFailureEventArgs(message, e));
```

Comment:
We should pass `message` as the source instead of `this` (i.e., it was the message that failed to decrypt).  This also kind of hides the wart of dealing with QueueMessage vs PeekedMessage behind the event handler pattern that forces `object sender`.